### PR TITLE
Convert cli tests into engine tests

### LINKED ISSUE #1226: Make cli integration tests into engine tests
The current cli tests are almost exclusively testing the `ert3.engine` module and that module has zero tests. This is technical debt that should be mitigated as the engine should have its own tests, and then the cli tests should probably just mock the engine all together.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #1226

### CODE REVIEW

File: tests/ert3/engine/integration/test_engine.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import pytest
+
+import ert3
+from ert3.console._console import _load_ensemble_config, _load_stages_config
+
+_EXAMPLES_ROOT = (
+    pathlib.Path(os.path.dirname(__file__)) / ".." / ".." / ".." / ".." / "examples"
+)
+_POLY_WORKSPACE_NAME = "polynomial"
+_POLY_WORKSPACE = _EXAMPLES_ROOT / _POLY_WORKSPACE_NAME
```
Comment (jondequinor) [inline tests/ert3/engine/integration/test_engine.py:16]: Have you considered providing these files in another way? Like attaching some test data to the test, or providing them via fixtures?

Comment (oyvindeide) [reply tests/ert3/engine/integration/test_engine.py:16]: Yes, will do that, this was just copying the existing tests set up. It is still relying a bit too much on files in magical locations...

**Review by jondequinor — APPROVED** [review]
LGTM as far as the moving operation goes…

**Review by jondequinor — APPROVED** [review]
Unconditional approve.

**Pull request merged**